libcontainer/container_linux.go

```diff
@@ -2237,6 +2296,28 @@ func (c *Container) bootstrapData(cloneFlags uintptr, nsMaps map[configs.Namespa
 		})
 	}
 
+	// Idmap mount sources to open.
+	if it == initStandard && c.shouldSendIdmapSources() {
+		var mounts []byte
+		for _, m := range c.config.Mounts {
+			if m.IsIDMap() {
+				// TODO: why do we need to duplicate this that is already done in
+				// libcontainer/specconv/spec_linux.go
```

Comment:
Does anyone know why we need this? I see it was added to avoid json strings to have null characters that will mix with the `byte(0)` we send a few lines below, the possible security issue and all...

But do we need to _duplicate_ this here? I just did it because in the case of `c.shouldSendMountSources()` was added there _too_. But isn't enough with the check done in spec_linux.go? Or if we check for this in the validator?

Reply:
This is because in case libcontainer is used by some other software (not runc), the `specconv` is no longer involved.

Reply:
@eiffel-fl can you please modify the comment above to explain why this is needed? 

In case libcontainer is used by some other software (not runc), the check we perform in  specconv is not executed, so we also check it here.

Reply:
Added a comment, thanks!